tests/Agent/IntegrationTests/UnboundedIntegrationTests/Postgres/PostgresExecuteScalarAsyncTests.cs

```diff
@@ -0,0 +1,194 @@
+﻿// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using MultiFunctionApplicationHelpers;
+using NewRelic.Agent.IntegrationTestHelpers;
+using NewRelic.Agent.IntegrationTestHelpers.Models;
+using NewRelic.Agent.IntegrationTests.Shared;
+using NewRelic.Testing.Assertions;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace NewRelic.Agent.UnboundedIntegrationTests.Postgres
+{
+    public abstract class PostgresSqlExecuteScalarAsyncTestsBase<TFixture> : NewRelicIntegrationTest<TFixture> where TFixture : ConsoleDynamicMethodFixture
+    {
+        private readonly ConsoleDynamicMethodFixture _fixture;
+
+        public PostgresSqlExecuteScalarAsyncTestsBase(TFixture fixture, ITestOutputHelper output) : base(fixture)
+        {
+            _fixture = fixture;
+            _fixture.TestLogger = output;
+
+            _fixture.AddCommand($"PostgresSqlExerciser ExecuteScalarAsync");
+
+            _fixture.Actions
+            (
+                setupConfiguration: () =>
+                {
+                    var configPath = fixture.DestinationNewRelicConfigFilePath;
+                    var configModifier = new NewRelicConfigModifier(configPath);
+
+                    configModifier.ForceTransactionTraces()
+                    .SetLogLevel("finest");
+
+                    CommonUtils.ModifyOrCreateXmlAttributeInNewRelicConfig(configPath, new[] { "configuration", "transactionTracer" }, "explainThreshold", "1");
+
+                    var instrumentationFilePath = $@"{fixture.DestinationNewRelicExtensionsDirectoryPath}\NewRelic.Providers.Wrapper.Sql.Instrumentation.xml";
+                    CommonUtils.SetAttributeOnTracerFactoryInNewRelicInstrumentation(instrumentationFilePath, "", "enabled", "true");
+                }
+            );
+
+            _fixture.AddActions(exerciseApplication: () => _fixture.AgentLog.WaitForLogLine(AgentLogBase.SqlTraceDataLogLineRegex, TimeSpan.FromMinutes(2)));
```

Comment:
There is a downside to waiting for this log regex. The test will take upwards of a minute until the default harvest cycle for sql traces elapses (60 seconds). 

As all these tests have been converted to include only one transaction, we should be able to wait for the TransactionFinalized log event which is logged as soon as the transaction ends and all data has been forwarded to  the appropriate aggregators. We make use of `SendDataOnExit` in our tests which causes a final harvest cycle to be invoked during application shutdown, which will send out all data staged in aggregators. This has the added benefit of not having a test wait around for a whole minute until the default harvest cycle comes around.